### ISSUE TITLE
IDE0055 tries to move a ternary operator a strange place

### ISSUE THREAD

**Issue by jasonmalinowski** [dotnet/roslyn#36839]
This line:

https://github.com/dotnet/roslyn/blob/e65f0cb2a986fc7176e5221e31521c804135c88b/src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs#L1051

is flagged by IDE0055. When applying the fix it wants to outdent that : a lot, which doesn't make any sense at all.

**Comment by jasonmalinowski** [comment]
FYI to @heejaechang and @sharwell.

**Comment by glen-84** [comment]
I'm not sure if it's the same thing, but in enabling IDE0055 for the Hot Chocolate repository, there was really only one issue (in 4 locations):

```cs
Func<ISyntaxNode, TContext, TContext> enterFunc = enter is not null
    ? (node, context) =>
    {
        context.Navigator.Push(node);
        return enter(node, context);
    }
    : (node, context) => // <-- It wants the colon here to be out-dented.
    {
        context.Navigator.Push(node);
        return context;
    };
```

![Image](https://github.com/user-attachments/assets/094422f4-cb7d-4ee3-8d4a-98cc436855ac)

![Image](https://github.com/user-attachments/assets/254a0e60-8bc0-4ac8-8d7a-d992fcca525e)
